pkg/apis/serving/v1beta1/service_validation_test.go

```diff
@@ -64,6 +70,96 @@ func TestServiceValidation(t *testing.T) {
 			},
 		},
 		want: nil,
+	}, {
+		name: "invalid annotation label",
+		r: &Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "valid",
+				Annotations: map[string]string{
+					"serving.knative.dev/testAnnotation": "value",
+				},
+			},
+			Spec: ServiceSpec{
+				ConfigurationSpec: goodConfigSpec,
+				RouteSpec:         goodRouteSpec,
+			},
+		},
+		want: apis.ErrInvalidValue("value", "metadata.annotations.serving.knative.dev/testAnnotation"),
+	}, {
+		name: "valid annotation label",
+		r: &Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "valid",
+				Annotations: map[string]string{
+					"testAnnotation": "testValue",
+				},
+			},
+			Spec: ServiceSpec{
+				ConfigurationSpec: goodConfigSpec,
+				RouteSpec:         goodRouteSpec,
+			},
+		},
+		want: nil,
+	}, {
+		name: "valid visibility label",
+		r: &Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "valid",
+				Labels: map[string]string{
+					"serving.knative.dev/visibility": "cluster-local",
+				},
+			},
+			Spec: ServiceSpec{
+				ConfigurationSpec: goodConfigSpec,
+				RouteSpec:         goodRouteSpec,
+			},
+		},
+		want: nil,
+	}, {
+		name: "invalid knative label",
+		r: &Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "valid",
+				Labels: map[string]string{
+					"serving.knative.dev/name": "some-value",
+				},
+			},
+			Spec: ServiceSpec{
+				ConfigurationSpec: goodConfigSpec,
+				RouteSpec:         goodRouteSpec,
+			},
+		},
+		want: apis.ErrInvalidKeyName("serving.knative.dev/name", "metadata.label"),
+	}, {
+		name: "valid non knative label",
+		r: &Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "valid",
+				Labels: map[string]string{
+					"serving.name": "some-name",
+				},
+			},
+			Spec: ServiceSpec{
+				ConfigurationSpec: goodConfigSpec,
+				RouteSpec:         goodRouteSpec,
+			},
+		},
+		want: nil,
+	}, {
+		name: "invalid visibility label value",
+		r: &Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "valid",
+				Labels: map[string]string{
+					"serving.knative.dev/visibility": "bad-label",
+				},
+			},
+			Spec: ServiceSpec{
+				ConfigurationSpec: goodConfigSpec,
+				RouteSpec:         goodRouteSpec,
+			},
+		},
+		want: apis.ErrInvalidValue("bad-label", "metadata.label.[serving.knative.dev/visibility]"),
```

Comment:
Please use constant `config.VisibilityLabelKey` instead of `serving.knative.dev/visibility`